src/compiler/checker.ts

```diff
@@ -22060,7 +22089,7 @@ namespace ts {
             forEachReturnStatement(<Block>func.body, returnStatement => {
                 const expr = returnStatement.expression;
                 if (expr) {
-                    let type = checkExpressionCached(expr, checkMode);
+                    let type = checkExpressionCached(expr, checkMode && checkMode && checkMode & ~CheckMode.SkipGenericFunctions);

```

Comment:
This line seems like a typo.

Reply:
I just wanted to be extra sure!

Reply:
I heard this is how to make really really thread-safe code in Node dot jay ess.